### PR TITLE
WIP Additional Modules for Achilles

### PR DESCRIPTION
**Pull request by uaon**
When merged this pull request will: Add 2 modules for Achilles
+ Power Outage module, this module allows the option of damaging all lightpoles / lamps within a radius to give the effect of a power outage in a city or otherwise.

+ Timer Countdown module, this gives the option of having a countdown without the mission ending as the vanilla module will end the mission when counter reaches 0

- Removed Tab indents from last pull request.
+ Added entries into fn_initModules.sqf and CfgFunctions.hpp for both modules to be added.

> Unsure of proper placement for modules in the Achilles tree. feel free to make changes as necessary to any / all parts of the modules. 

> Params seemed to consistently give errors when attempted with the Power outage module, Defined variables by type instead of inside a parameter. further revision might be necessary.

### CODE REVIEW

File: cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    "Power Outage",
+    [
+        ["Meters","", 1200]
+    ]
+] call Ares_fnc_showChooseDialog;
+  if (count _dialogResult == 0) exitWith {};
+
+  private _OutageRadius = _dialogResult select 0;
+  private _OutageRadius = parseNumber _OutageRadius;
+  Private _OutageArray = [
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: Possible add all cup lamps as well?

File: cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+ * this call cScripts_fnc_modulePowerOutage
+ *
+ * Public: No
+ */
+
+private _center = position _logic;
+private _OutageArray = [];
+private _OutageRadius = [""];
+
+    _dialogResult = [
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: weird indent going on here.

File: cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+        ["Seconds","", "60"]
+    ]
+] call Ares_fnc_showChooseDialog;
+
+if (count _dialogResult == 0) exitWith {};
+
+
+private _Ctimer = _dialogResult select 0;
+
+        _time = [] call bis_fnc_countdown;
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf]: missing private as well as weird indent

File: cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+    "Countdown Timer",
+    [
+        ["Seconds","", "60"]
+    ]
+] call Ares_fnc_showChooseDialog;
+
+if (count _dialogResult == 0) exitWith {};
+
+
+private _Ctimer = _dialogResult select 0;
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf]: is already private due to params.

File: cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+
+        _time = [] call bis_fnc_countdown;
+        if (_time <= 0) then {_time = estimatedendservertime - servertime;};
+        _value = _time call compile _Ctimer;
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf]: missing private

File: cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf (hunk 1 above)
Comment (uaon) [reply cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: I'll check through Cups terrains / ..mod and see if I can find a class list for those to add into the array. Also from further testing it appears Fuel stations are exempt. still working out what is needed for those as well.

**Comment by uaon** [comment]
@ColdEvul Made changes to files as per your provided guidance,
+ Added ability to switch power on/off lights using `BIS_fnc_switchLamp;` to PowerOutage module
+ Set private variables in params or used `private`
+ removed several excessive indents, should appear more pleasing.

File: cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+ * 
+ * Arguments:
+ * 0: None
+ *
+ * Example:
+ * this call cScripts_fnc_modulePowerOutage
+ *
+ * Public: No
+ */
+[[], [""], []] call {
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: what is that??? O.o

File: cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    _valueText = format ["%1 %2",ceil (_value / 60),localize "STR_A3_RSCMPPROGRESS_min"];
+    [["Countdown",[_valueText]],"bis_fnc_showNotification"] call bis_fnc_mp;
+};
+            
+if (ismultiplayer) then {
+    estimatedtimeleft _value;
+} else {
+    _value call bis_fnc_countdown;
+    };
+} else {
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf]: this is weirdly set up and will cause errors

File: cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+    "Land_lampa_ind_b",
+    "Land_lampa_sidl",
+    "Land_PowLines_Conc2L_EP1",
+    "Land_Lamp_Street1_EP1",
+    "Land_Lamp_Street2_EP1",
+    "Land_Lamp_Small_EP1",
+    "Land_Lampa_Ind_EP1"
+];
+{
+[_x, _pstate] call BIS_fnc_switchLamp;
```
Comment (ColdEvul) [inline cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: slap a indent here for readability

Comment (ColdEvul) [reply cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: Like this:
```suggestion
{
    //_x setHit ["light_1_hitpoint", 0.97]; 
    //_x setHit ["light_2_hitpoint", 0.97]; 
    //_x setHit ["light_3_hitpoint", 0.97]; 
    //_x setHit ["light_4_hitpoint", 0.97];
} forEach nearestObjects [_center, _OutageArray, _OutageRadius];
```

File: cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf (hunk 6 above)
Comment (ColdEvul) [reply cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: this is enough;
```suggestion
params ["_OutageRadius", "_OutageArray", "_center"];
```

Comment (ColdEvul) [reply cScripts/CavFnc/functions/modules/fn_modulePowerOutage.sqf]: you can also do like this:
```suggestion
params [
    ["_OutageRadius", 0, [0]], 
    ["_OutageArray", 0, [0]],
    ["_center", 0, [0]]
];
```
To add a **default** and **expected value**.
0 = number
"" = for string

File: cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf (hunk 7 above)
Comment (ColdEvul) [reply cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf]: this in particular is not due to indent its due to the setup of the if sats.
For default structure of if sats here for used here is the following format:
```
if (_condition) then {
    //Code
} else {
    //Code
};
if (_condition) then { /*code*/};
```

But above you have done like this witch make no sence:
```
if (ismultiplayer) then {
    estimatedtimeleft _value;
} else {
    _value call bis_fnc_countdown;
    };
} else {
if (ismultiplayer) then {
    estimatedtimeleft 1e10;
} else {
    -1 call bis_fnc_countdown;
};
```
line 42
starts with `}; } else {`

Comment (ColdEvul) [reply cScripts/CavFnc/functions/modules/fn_moduleCreateTimer.sqf]: think I see what happend here now it's the indent. I run like this:
```
if () then {
    if () then {
        if () then {
        } 
    } else {
    };
} ;

```

**Pull request closed without merging**